### PR TITLE
Refactor pkg/kubelet/kubelet.go: probeContainer().

### PR DESCRIPTION
**Pull request by yifan-gu**
Split the function into probeContainerLiveness() and
probeContainerReadiness(). Also add tests.

### CODE REVIEW

**Comment by yifan-gu** [comment]
This refactor just makes sure the semantics for probeContainer does not break.
But maybe we want to take a chance to revisit it? Like should we set readiness when Liveness probe returns error?

**Comment by yifan-gu** [comment]
Will run the e2e and report.

File: pkg/kubelet/probe.go [hunk 1]
```diff
@@ -24,53 +24,109 @@ import (
...
+		}
+	}
+	return result, err
+}
+
+// probeContainerLiveness probes the readiness of a container.
+// If the initial delay on the readiness probe has not passed the probe will return probe.Failure.
+func (kl *Kubelet) probeContainerReadiness(pod *api.BoundPod, status api.PodStatus, container api.Container, dockerContainer *docker.APIContainers) (probe.Result, error) {
+	p := container.ReadinessProbe
+	fmt.Println(p)
```
Comment (vmarmol) [inline pkg/kubelet/probe.go]: debugging?

File: pkg/kubelet/probe.go [hunk 2]
```diff
@@ -24,53 +24,109 @@ import (
...
+		glog.Warningf("No ref for pod '%v' - '%v'", containerID, container.Name)
+	} else {
+		kl.recorder.Eventf(ref, "unhealthy", "Liveness Probe Failed %v - %v", containerID, container.Name)
+	}
+	return ready, err
+}
+
+// probeContainerLiveness probes the liveness of a container.
+// If the initalDelay since container creation on liveness probe has not passed the probe will return probe.Success.
+func (kl *Kubelet) probeContainerLiveness(pod *api.BoundPod, status api.PodStatus, container api.Container, dockerContainer *docker.APIContainers) (probe.Result, error) {
```
Comment (vmarmol) [inline pkg/kubelet/probe.go]: I think we can collapse these back into one function that takes the probe and the defaultResult. The code is near identical.

**Comment by vmarmol** [comment]
Minor nits, this looks good.

File: pkg/kubelet/probe.go (hunk 1 above)
Comment (yifan-gu) [inline pkg/kubelet/probe.go]: oops

File: pkg/kubelet/probe.go (hunk 2 above)
Comment (mikedanese) [inline pkg/kubelet/probe.go]: deja vu :)
https://github.com/GoogleCloudPlatform/kubernetes/pull/4048#discussion_r24036442

Comment (yifan-gu) [inline pkg/kubelet/probe.go]: I would argue that the old function that takes the `defaultResult` is rather confusing. That looks like a generic function at first glance, but indeed it reduces the repeatness by hurting the readability IMO

Comment (vmarmol) [inline pkg/kubelet/probe.go]: How about we meet in the middle and keep the functions separate but do the first if and the for loop in a helper? That way the defaultResult part is called out explicitly and the meaty heart of the function is re-used :)

Comment (vmarmol) [inline pkg/kubelet/probe.go]: I mean just the loop

Comment (yifan-gu) [inline pkg/kubelet/probe.go]: @vmarmol :+1: What function name will you suggest? tryRunProbe?

Comment (yifan-gu) [inline pkg/kubelet/probe.go]: @mikedanese Ah... Thanks for the reference.

Comment (vmarmol) [inline pkg/kubelet/probe.go]: LOL, I'm the worst person for names but: runProbeWithRetries()?

Comment (yifan-gu) [inline pkg/kubelet/probe.go]: Yet seems to be better than me.
Done.

**Comment by vmarmol** [comment]
LGTM, thanks @yifan-gu! Will merge on green

**Pull request merged**